paper.tex

```diff
@@ -1,300 +1,290 @@
+
+\subsection{DevOps practices}
+
+The DevOps methods emphasize on exercising Quality Assurance techniques to avoid infrastructure disruption whenever new developments are deployed into production. The INDIGO--DataCloud project promoted the application of a CI scenario to enforce the Quality Assurance requirements for any piece of software being produced within the project. Such environment requires an automated ready--to--go infrastructure where the different services involved (source code management platform, automation server) interact with each other to trigger the quality check execution and return back the exit codes, together with their outputs. To accomplish this, the project leveraged tightly integrated open source tools such as GitHub and Jenkins, as well as relying on Docker container provisioning to speed up the source code validation. This CI approach is guiding the project’s software development phase throughout the first and second major releases.
+
+As DevOps suggests, frequent releases positively affects the reliability of the software. The software updates of INDIGO--DataCloud products taking place since the second major release are passing through a Continuous Delivery (CD) pipeline that adds the packaging of the software right after the execution of the quality checks (as part of the CI). The steps defined within the CD pipeline differ whether the software is to be distributed in the form of Docker images or via operating system’s packages. In the latter case an extra validation step is added, consisting in the product’s deployment using a Configuration Management (CM) solution. The installation uses the packages being built in the previous step, which are being uploaded to a testing repository. In the case of Docker images, the CM is used to build the image itself, installing and configuring the product before uploading it to the DockerHub repository \cite{indigo-dockerhub}. Figure \ref{fig:fig_CD} showcases the workflow followed by the products distributed as Docker images.
+
+\begin{figure*}
+\centering
+\includegraphics[width=\textwidth]{images/devops.png}
+\caption{Continuous Delivery workflow for Docker images.}
+\label{fig:fig_CD}
+\end{figure*}
+
+\subsection{Software Quality Procedures}
+
+The software quality procedures \cite{indigo-d31} cover; 1) the identification and description of the \emph{quality requirements} that the software need to comply with, and 2) the \emph{quality metrics}.
```

Comment:
semicolon after cover?

Reply:
seemed ok after its kind of itemized, just because of that